src/Server.cpp

```diff
@@ -38,14 +38,20 @@ Server::Server(const std::string &address, const uint16_t port,
 
 Server::~Server() { std::cout << "Bye~" << std::endl; }
 
+Server *Server::GetInstance(const std::string &address, const uint16_t port,
+                            const std::size_t max_session_count,
+                            const std::size_t max_thread_count) {
+  static Server instance(address, port, max_session_count, max_thread_count);
```

Comment:
[Optional] You can consider splitting GetInstance(...) into InitInstance(...) and GetInstance().

InitInstance(...) will take all the constructor arguments (address, port... etc), so that when we use GetInstance(), we'll not need those arguments.

The pointer to the singleton can be placed in an anonymous namespace instead of being a static variable, so that both function can access them.